# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/io/nsLocalFileWin.h
# Commit: 25b264ceed3d
# Full Hash: 25b264ceed3d448fb2c8cbee0662baa03303b48e
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2025-10-06 08:04 +0000
# Regressor Bug: 1980886
# File Overlap Count: 1
# Overlapping Files: xpcom/io/nsLocalFileWin.cpp
# Description:
#   Bug 1980886 p1 - Refactor CopySingleFile to better reflect its usage and behaviour. r=yjuglaret
# ==============================================================================

diff -r 4a26473354f3 -r 25b264ceed3d xpcom/io/nsLocalFileWin.h
--- a/xpcom/io/nsLocalFileWin.h	Mon Oct 06 10:48:11 2025 +0300
+++ b/xpcom/io/nsLocalFileWin.h	Mon Oct 06 08:04:24 2025 +0000
@@ -108,8 +108,15 @@
 
   nsresult CopyMove(nsIFile* aNewParentDir, const nsAString& aNewName,
                     uint32_t aOptions);
-  nsresult CopySingleFile(nsIFile* aSource, nsIFile* aDest,
-                          const nsAString& aNewName, uint32_t aOptions);
+  /**
+   * Moves or copies this file or directory as a single entity.
+   * @param aDest the destination directory
+   * @param aNewName optional new name
+   * @param aOptions CopyFileOption flags
+   */
+  nsresult MoveOrCopyAsSingleFileOrDir(nsIFile* aDest,
+                                       const nsAString& aNewName,
+                                       uint32_t aOptions);
 
   enum class TimeField { AccessedTime, ModifiedTime };
 
